### PR TITLE
KAFKA-12180: Implement the KIP-631 message generator changes

### PR DESCRIPTION
**Pull request by cmccabe**
* Implement the uint16 type
* Implement MetadataRecordType and MetadataJsonConverters

### CODE REVIEW

File: generator/src/main/java/org/apache/kafka/message/MessageSpecType.java [hunk 1]
```diff
@@ -30,5 +30,8 @@
     HEADER,
 
     @JsonProperty("data")
-    DATA;
+    DATA,
+
+    @JsonProperty("metadata")
```
Comment (hachikuji) [inline generator/src/main/java/org/apache/kafka/message/MessageSpecType.java:35]: Might be helpful to have a short description here (or somewhere else if you can think of a better location) which explains that this type is reserved for the controller record types.

**Review by hachikuji — APPROVED** [review]
Thanks, LGTM. Minor suggestion to document the metadata spec type.

File: generator/src/main/java/org/apache/kafka/message/MessageSpecType.java (hunk 1 above)
Comment (cmccabe) [reply generator/src/main/java/org/apache/kafka/message/MessageSpecType.java:35]: I added a comment

**Comment by cmccabe** [comment]
merged, thanks!

**Pull request closed without merging**